micro-rdk/src/esp32/conn/wifi_error.rs

```diff
@@ -0,0 +1,171 @@
+            wifi_err_reason_t_WIFI_REASON_MISSING_ACKS => Self::MissingAcks(v),
+            wifi_err_reason_t_WIFI_REASON_EXCEEDED_TXOP => Self::ExceededTxOp(v),
+            wifi_err_reason_t_WIFI_REASON_STA_LEAVING => Self::StaLeaving(v),
+            wifi_err_reason_t_WIFI_REASON_END_BA => Self::EndBA(v),
+            wifi_err_reason_t_WIFI_REASON_UNKNOWN_BA => Self::UnknownBA(v),
+
+            wifi_err_reason_t_WIFI_REASON_TIMEOUT => Self::Timeout(v),
+            wifi_err_reason_t_WIFI_REASON_PEER_INITIATED => Self::PeerInitiated(v),
+            wifi_err_reason_t_WIFI_REASON_AP_INITIATED => Self::ApInitiated(v),
+            wifi_err_reason_t_WIFI_REASON_INVALID_FT_ACTION_FRAME_COUNT => {
+                Self::InvalidFtActionFrameCount(v)
+            }
+            wifi_err_reason_t_WIFI_REASON_INVALID_PMKID => Self::InvalidPmkid(v),
+            wifi_err_reason_t_WIFI_REASON_INVALID_MDE => Self::InvalidMde(v),
+            wifi_err_reason_t_WIFI_REASON_INVALID_FTE => Self::InvalidFte(v),
+
+            wifi_err_reason_t_WIFI_REASON_TRANSMISSION_LINK_ESTABLISH_FAILED => {
+                Self::TransmissionLinkEstablishFailed(v)
+            }
+            wifi_err_reason_t_WIFI_REASON_ALTERATIVE_CHANNEL_OCCUPIED => {
+                Self::AlternativeChannelOccupied(v)
+            }
+            wifi_err_reason_t_WIFI_REASON_BEACON_TIMEOUT => Self::BeaconTimeout(v),
+            wifi_err_reason_t_WIFI_REASON_NO_AP_FOUND => Self::NoApFound(v),
+            wifi_err_reason_t_WIFI_REASON_AUTH_FAIL => Self::AuthFail(v),
+            wifi_err_reason_t_WIFI_REASON_ASSOC_FAIL => Self::AssocFail(v),
+            wifi_err_reason_t_WIFI_REASON_HANDSHAKE_TIMEOUT => Self::HandshakeTimeout(v),
+            wifi_err_reason_t_WIFI_REASON_CONNECTION_FAIL => Self::ConnectionFail(v),
+            wifi_err_reason_t_WIFI_REASON_AP_TSF_RESET => Self::ApTsfReset(t),
+            wifi_err_reason_t_WIFI_REASON_ROAMING => Self::Roaming(v),
+            wifi_err_reason_t_WIFI_REASON_ASSOC_COMEBACK_TIME_TOO_LONG => {
+                Self::AssocComebackTimeTooLong(v)
+            }
+            wifi_err_reason_t_WIFI_REASON_SA_QUERY_TIMEOUT => Self::SaQueryTimeout(v),
+
+            wifi_err_reason_t_WIFI_REASON_NO_AP_FOUND_W_COMPATIBLE_SECURITY => {
+                Self::NoApFoundWithCompatibleSecurity(v)
+            }
+            wifi_err_reason_t_WIFI_REASON_NO_AP_FOUND_IN_AUTHMODE_THRESHOLD => {
+                Self::NoApFoundInAuthModeThreshold(v)
+            }
+            wifi_err_reason_t_WIFI_REASON_NO_AP_FOUND_IN_RSSI_THRESHOLD => {
+                Self::NoApFoundInRssiThreshold(v)
+            }
+            _ => Self::Unrecognized(v),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub enum WifiErrReason {
+    Unspecified(c_uint),
+    AuthExpire(c_uint),
+    AuthLeave(c_uint),
+    AssocExpire(c_uint),
+    AssocTooMany(c_uint),
+    NotAuthed(c_uint),
+    NotAssoced(c_uint),
+    AssocLeave(c_uint),
+    AssocNotAuthed(c_uint),
```

Comment:
I have to wonder if there isn't some macro crate out there that can emit this kind of boilerplate for us given a C enum.

Also surprised that esp-idf-{svc,sys} doesn't do this for us.